Services/Component/classes/class.ilPluginAdmin.php

```diff
@@ -243,8 +242,9 @@ protected function parsePluginPhp($plugin_php_file)
             "ilias_max_version" => $ilias_max_version,
             "responsible" => $responsible,
             "responsible_mail" => $responsible_mail,
-            "learning_progress" => (bool) $learning_progress,
-            "supports_export" => (bool) $supports_export,
+            "learning_progress" => (bool) ($learning_progress ?? false),
+            "supports_export" => (bool) ($supports_export ?? false),
+            "supports_cli_setup" => (bool) ($supports_cli_setup ?? true)
```

Comment:
I know we had discussed this and probably the JF decision is the same. but i currently tend to do opt-in instead of opt-out. as default 'false'.

i am afraid that otherwise a lot of plugins will appear relatively quickly where this is not possible and so updates to ILIAS 7 can quickly become a problem. developers of plugins could first check with opt-in that everything works as desired before this is offered. what do you think?

Reply:
I disagree. Maybe we would want to discuss this on the JF with other developers. For one, I wonder why and how many plugins would really not be installable with this mechanism. I understood that you are using GUIs in the `afterInstallation`-hook of the plugins, but is that really common for many plugins and are there other failure modes? After all, the CLI setup uses the same mechanism as the GUI for plugins. Second, developers need to test the plugins and bump the version in the `plugin.php` anyway. There could be some problem regarding the general information about updates for plugin developers. I don't think, we have a strategy here, right? Finally I think this is just one step on the way to move plugins and the other components closer together, at least on a technical base. So this (defaulting this to `true`) is something that will happen eventually anyway. I don't see enough reasons to defer this.

Since this is something we can change very cheaply, I think we could move on with this PR anyway (right?) and, if you want, bring this up on the JF. I won't attend on 20-11-09, but I'll be there the one afterwards.

Reply:
@klees the here given fallback would be also very usefull ot apply to ILIAS6 to prevent notices inside the log on unaccessable variables.